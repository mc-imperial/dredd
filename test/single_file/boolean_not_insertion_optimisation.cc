int main() {
  bool x = true;
  x = false;
}
