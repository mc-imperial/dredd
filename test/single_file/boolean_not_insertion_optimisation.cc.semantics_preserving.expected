#include <cinttypes>
#include <cstddef>
#include <functional>
#include <string>


#ifdef _MSC_VER
#define thread_local __declspec(thread)
#elif __APPLE__
#define thread_local __thread
#endif

#include <limits>
#include <cmath>
static thread_local unsigned long long int no_op = 0;

#define MUTATION_RETURN(arg) actual_result
#define REPLACE_EXPR_TRUE(args) if ((args) != actual_result) no_op++
#define REPLACE_EXPR_FALSE(args) if ((args) != actual_result) no_op++
#define MUTATION_PRELUDE(arg,type) type actual_result = (arg)
static bool __dredd_replace_expr_bool_true(bool arg, int local_mutation_id) {
  MUTATION_PRELUDE(arg,bool);
  REPLACE_EXPR_FALSE(false);
  return MUTATION_RETURN(arg);
}

static bool __dredd_replace_expr_bool_false(bool arg, int local_mutation_id) {
  MUTATION_PRELUDE(arg,bool);
  REPLACE_EXPR_TRUE(true);
  return MUTATION_RETURN(arg);
}

int main() {
  bool x = __dredd_replace_expr_bool_true(true, 0);
  x = __dredd_replace_expr_bool_false(false, 1);
}
